podcast_agent/client/transcriber.py

```diff
@@ -0,0 +1,43 @@
+class Transcriber:
+    def __init__(self):
+        import openai
+        import os
+        openai.api_key = os.environ['OPENAI_API_KEY']
+    
+    #TODO different path for large files
+    #TODO read from cloud
+    def read_from_url(self, url):
+        #wrapper around openai whisper transcription service to read from 
+        # given url and returns transcription
+        #
+        #url: location of audio file
+        #returns: text transcription of file
+
+        import openai
+        import os
+
+        if os.path.exists(url):
+            transcript = self.local_file_to_transcription(open(url, "rb"))
+        
+        elif url.startswith('https://'):
+            import tempfile
+            if url.startswith('https://www.listennotes'):
+                import requests
+                audio_file = requests.get(url)
+
+                with tempfile.NamedTemporaryFile(suffix='.mp3', mode='rb+', delete=True) as temp_file:
+                    temp_file.write(audio_file.content)
```

Comment:
@jonmc12 i think we cannot directly pass in-memory bytes to `openai.Audio.transcribe` without changing library code as discussed in this [thread](https://github.com/openai/whisper/discussions/908) so this workaround writes a temp file and deletes it later